crates/rpc/rpc/src/eth/api/mod.rs

```diff
@@ -119,21 +150,21 @@ where
     pub(crate) fn state_at_block_id_or_latest(
         &self,
         block_id: Option<BlockId>,
-    ) -> Result<Option<<Client as StateProviderFactory>::HistorySP<'_>>> {
+    ) -> Result<Option<HistoryOrLatest<'_, Client>>> {
```

Comment:
Why is this change needed? `latest_state()` did the same before, no?

Reply:
Yes, but this way the consumer of the `Factory` doesn't need to explicitly use `LatestSP`, which would need an additional check so he knows if the `BlockId` received references the latest block of the chain or not. This check wasn't being done before, and we were always using the `HistorySP` by default.